test/BenchmarkConfigurationTest.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Andre Schlegel (April of 2023,
+// schlegea@informatik.uni-freiburg.de)
+
+#include <gtest/gtest.h>
+
+#include "../benchmark/infrastructure/BenchmarkConfiguration.h"
+
+TEST(BenchmarkConfigurationTest, ParseJsonTest) {
+  ad_benchmark::BenchmarkConfiguration config{};
+
+  // The json string for testing `setJsonString`.
+  // The content is a modified demo from
+  // `https://www.objgen.com/json?demo=true`.
+  const std::string testJsonString{R"--({
+  "product": "Live JSON generator",
+  "version": 3.1,
+  "demo": true,
+  "person": {
+    "id": 12345,
+    "name": "John Doe",
+    "phones": {
+      "home": "800-123-4567",
+      "mobile": "877-123-1234"
+    },
+    "email": [
+      "jd@example.com",
+      "jd@example.org"
+    ]
+  }
+  })--"};
+
+  // Set and checking some samples.
+  config.setJsonString(testJsonString);
+
+  auto checkSamples = [&config]() {
+    ASSERT_EQ((float)3.1,
+              config.getValueByNestedKeys<float>("version").value());
+    ASSERT_EQ(std::string{"Live JSON generator"},
+              config.getValueByNestedKeys<std::string>("product").value());
+    ASSERT_EQ(true, config.getValueByNestedKeys<bool>("demo").value());
+    ASSERT_EQ(
+        std::string{"800-123-4567"},
+        config.getValueByNestedKeys<std::string>("person", "phones", "home")
+            .value());
+    ASSERT_EQ(
+        std::string{"jd@example.org"},
+        config.getValueByNestedKeys<std::string>("person", "email", 1).value());
+  };
+  checkSamples();
```

Comment:
What happens, when the keys are valid, but the type is a mismatch?
This should be tested, and this case should also yield a reasonable execpetion message (not only JSON::internal_error).

Reply:
Done, see [6a4a814](https://github.com/ad-freiburg/qlever/pull/860/commits/6a4a814dcfb7265138d2b9224ca2ef31c9473af1).